src/bot.coffee

```diff
@@ -80,10 +81,13 @@ class SlackBot extends Adapter
   # @param {...(string|Object)} messages - fully documented in SlackClient
   ###
   send: (envelope, messages...) ->
-    # TODO: if the sender is interested in the completion, the last item in `messages` will be a function
+    callback = ->
+    if typeof(messages[messages.length - 1]) == 'function'
+      callback = messages.pop()
     for message in messages
       # NOTE: perhaps do envelope manipulation here instead of in the client (separation of concerns)
       @client.send(envelope, message) unless message is ""
+    Promise.all(messages).then(callback.bind(null, null), callback)
```

Comment:
i'm not sure how this works. `messages` is an array of either a string or an object, which might have a callback as the last item (after this change). i think what you meant is to collect all the Promises returned from `@client.send(...)`. maybe the for loop needs to be refactored into a something like `messagePromises = messages.map(...)`, so that this line can become `Promise.all(messagePromises)`. you also need to be careful in the `messages.map(...)` mapping function to skip operating on any item that looks like a function, probably by just returning a simple value or `Promise.resolve()`.